cmd/maestro/server/grpc_server.go

```diff
@@ -150,8 +176,22 @@ func (svr *GRPCServer) Publish(ctx context.Context, pubReq *pbv1.PublishRequest)
 			return nil, fmt.Errorf("failed to update resource: %v", err)
 		}
 	case common.DeleteRequestAction:
-		err := svr.resourceService.MarkAsDeleting(ctx, res.ID)
+		res, err = decode(eventType.CloudEventsDataType, evt)
 		if err != nil {
+			return nil, fmt.Errorf("failed to decode cloudevent: %v", err)
+		}
+		var deletionTimestamp time.Time
+		evtExtensions := evt.Context.GetExtensions()
+		if _, ok := evtExtensions[codec.ExtensionWorkMeta]; ok {
+			if deletionTimestampValue, exists := evtExtensions[types.ExtensionDeletionTimestamp]; exists {
```

Comment:
should we ignore the client deletion timestamp? instead of setting the deletion timestamp on the server

Reply:
done, use server generated deletion timestamp.